test/integration/backward_compatible/parallel/serialization/compact/CompactPublicAPIsTest.js

```diff
@@ -0,0 +1,952 @@
+                    maxIdleSeconds: 2,
+                    inMemoryFormat: 'BINARY',
+                    timeToLiveSeconds: 3,
+                    evictionPolicy: 'lru',
+                    evictionMaxSize: 3000,
+                    evictionSamplingCount: 4,
+                    evictionSamplingPoolSize: 8
+                }
+            }
+        };
+
+        client = await testFactory.newHazelcastClientForParallelTests(clientConfig, member);
+        map = await client.getMap(name);
+        nearCachedMap1 = await client.getMap('nearCached1' + name);
+        nearCachedMap2 = await client.getMap('nearCached2' + name);
+        multimap = await client.getMultiMap(name);
+        replicatedMap = await client.getReplicatedMap(name);
+        list = await client.getList(name);
+        atomicReference = await client.getCPSubsystem().getAtomicReference(name);
+        queue = await client.getQueue(name);
+        set = await client.getSet(name);
+        topic = await client.getReliableTopic(name);
+        ringBuffer = await client.getRingbuffer(name);
+        compactSerializerUsed = false;
+    });
+
+    afterEach(async function () {
+        compactSerializerUsed.should.be.true;
+        sandbox.restore();
+        await map.destroy();
+        await nearCachedMap1.destroy();
+        await nearCachedMap2.destroy();
+        await multimap.destroy();
+        await replicatedMap.destroy();
+        await list.destroy();
+        await atomicReference.destroy();
+        await queue.destroy();
+        await set.destroy();
+        await topic.destroy();
+        await ringBuffer.destroy();
+        await testFactory.shutdownAllClients();
+    });
+
+    after(async function () {
+        await testFactory.shutdownAll();
+    });
+
+    const putToMapFromOtherClient = async (mapName, key, value) => {
+        const client = await testFactory.newHazelcastClientForParallelTests(clientConfig, member);
+        const map = await client.getMap(mapName);
+        await map.put(key, value);
+    };
+
+    const putToReplicatedMapFromOtherClient = async (replicatedMapName, key, value) => {
+        const client = await testFactory.newHazelcastClientForParallelTests(clientConfig, member);
+        const replicatedMap = await client.getReplicatedMap(replicatedMapName);
+        await replicatedMap.put(key, value);
+    };
+
+    const assertEntryEvent = async (events) => {
+        await TestUtil.assertTrueEventually(async () => {
+            events.length.should.be.equal(1);
+            const event = events[0];
+            event.key.should.be.deep.equal(INNER_INSTANCE);
+            event.value.should.be.deep.equal(OUTER_INSTANCE);
+            should.equal(event.oldValue, null);
+            should.equal(event.mergingValue, null);
+        });
+    };
+
+    describe('Map/NearCaches', function () {
+        it('aggregate', async function() {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
```

Comment:
instead of a loop like this in each test method body, can we do that once in the describe block?

Reply:
Since we instantiate data structures before each test, that is not possible. I don't it will work even if we do it in `before()` because first thing mocha does is to collect tests and then run before().